File: SocialPulse/SocialPulse.Core/Entities/Comment.cs
﻿namespace SocialPulse.Core
{
    public class Comment : BaseEntity
    {
        public string Text { get; set; } = null!;

        public int PostId { get; set; }
        public Post Post { get; set; } = null!;
        public int UserId { get; set; }
        public User User { get; set; } = null!;
    }
}

File: SocialPulse/UI/admin_socialpulse/lib/models/comment.dart
import 'package:admin_socialpulse/models/post.dart';
import 'package:json_annotation/json_annotation.dart';
import 'package:admin_socialpulse/models/user.dart';

part 'comment.g.dart';

@JsonSerializable()
class Comment {
  int? id;
  String? text;
  int? userId;
  User? user;
  int? postId;
  Post? post;

  Comment(this.id, this.text, this.userId, this.user, this.postId, this.post);

  factory Comment.fromJson(Map<String, dynamic> json) =>
      _$CommentFromJson(json);

  Map<String, dynamic> toJson() => _$CommentToJson(this);
}
